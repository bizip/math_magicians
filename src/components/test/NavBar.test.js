import React from 'react';
import renderer from 'react-test-renderer';
import { HashRouter as Router } from 'react-router-dom';
import Navbar from '../../layouts/NavBar';

describe('Test for the navigation bar section', () => {
  it('It should render the navigation bar', () => {
    const nav = renderer
      .create(<Router><Navbar /></Router>)
      .toJSON();
    expect(nav).toMatchSnapshot();
  });
});
