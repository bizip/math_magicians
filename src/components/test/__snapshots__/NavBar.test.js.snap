// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` test if the navbar is rendered 1`] = `
<nav>
  <h1>
    Math magicians
  </h1>
  <ul
    className="nav-links"
  >
    <li>
      <a
        aria-current="page"
        className="active"
        href="#/"
        onClick={[Function]}
      >
        Home
      </a>
    </li>
    <li>
      <a
        className=""
        href="#/Calculator"
        onClick={[Function]}
      >
        Calculator
      </a>
    </li>
    <li>
      <a
        className=""
        href="#/quotes"
        onClick={[Function]}
      >
        Quotes
      </a>
    </li>
  </ul>
</nav>
`;

exports[`Test  test if the navbar is rendered 1`] = `
<nav>
  <h1>
    Math magicians
  </h1>
  <ul
    className="nav-links"
  >
    <li>
      <a
        aria-current="page"
        className="active"
        href="#/"
        onClick={[Function]}
      >
        Home
      </a>
    </li>
    <li>
      <a
        className=""
        href="#/Calculator"
        onClick={[Function]}
      >
        Calculator
      </a>
    </li>
    <li>
      <a
        className=""
        href="#/quotes"
        onClick={[Function]}
      >
        Quotes
      </a>
    </li>
  </ul>
</nav>
`;

exports[`Test for the navigation bar section  1`] = `
<nav>
  <h1>
    Math magicians
  </h1>
  <ul
    className="nav-links"
  >
    <li>
      <a
        aria-current="page"
        className="active"
        href="#/"
        onClick={[Function]}
      >
        Home
      </a>
    </li>
    <li>
      <a
        className=""
        href="#/Calculator"
        onClick={[Function]}
      >
        Calculator
      </a>
    </li>
    <li>
      <a
        className=""
        href="#/quotes"
        onClick={[Function]}
      >
        Quotes
      </a>
    </li>
  </ul>
</nav>
`;

exports[`Test for the navigation bar section It should render the navigation  1`] = `
<nav>
  <h1>
    Math magicians
  </h1>
  <ul
    className="nav-links"
  >
    <li>
      <a
        aria-current="page"
        className="active"
        href="#/"
        onClick={[Function]}
      >
        Home
      </a>
    </li>
    <li>
      <a
        className=""
        href="#/Calculator"
        onClick={[Function]}
      >
        Calculator
      </a>
    </li>
    <li>
      <a
        className=""
        href="#/quotes"
        onClick={[Function]}
      >
        Quotes
      </a>
    </li>
  </ul>
</nav>
`;

exports[`Test for the navigation bar section It should render the navigation bar 1`] = `
<nav>
  <h1>
    Math magicians
  </h1>
  <ul
    className="nav-links"
  >
    <li>
      <a
        aria-current="page"
        className="active"
        href="#/"
        onClick={[Function]}
      >
        Home
      </a>
    </li>
    <li>
      <a
        className=""
        href="#/Calculator"
        onClick={[Function]}
      >
        Calculator
      </a>
    </li>
    <li>
      <a
        className=""
        href="#/quotes"
        onClick={[Function]}
      >
        Quotes
      </a>
    </li>
  </ul>
</nav>
`;

exports[`Test for the navigation bar section test if the navbar is rendered 1`] = `
<nav>
  <h1>
    Math magicians
  </h1>
  <ul
    className="nav-links"
  >
    <li>
      <a
        aria-current="page"
        className="active"
        href="#/"
        onClick={[Function]}
      >
        Home
      </a>
    </li>
    <li>
      <a
        className=""
        href="#/Calculator"
        onClick={[Function]}
      >
        Calculator
      </a>
    </li>
    <li>
      <a
        className=""
        href="#/quotes"
        onClick={[Function]}
      >
        Quotes
      </a>
    </li>
  </ul>
</nav>
`;

exports[`Test of navbar test if the navbar is rendered 1`] = `
<nav>
  <h1>
    Math magicians
  </h1>
  <ul
    className="nav-links"
  >
    <li>
      <a
        aria-current="page"
        className="active"
        href="#/"
        onClick={[Function]}
      >
        Home
      </a>
    </li>
    <li>
      <a
        className=""
        href="#/Calculator"
        onClick={[Function]}
      >
        Calculator
      </a>
    </li>
    <li>
      <a
        className=""
        href="#/quotes"
        onClick={[Function]}
      >
        Quotes
      </a>
    </li>
  </ul>
</nav>
`;
